app/views/messages/_message.html.erb

```diff
@@ -1,284 +1,274 @@
-<%# locals: (message:, scroll_down: false, only_scroll_down_if_was_bottom: false, request_id: nil, timestamp: nil, thinking: nil, streamed: false, message_counter: -1) -%>
+<%# locals: (message:, request_id: nil, scroll_down: false, only_scroll_down_if_was_bottom: false, thinking: nil, streamed: false, message_counter: -1) -%>
 <%# request_id is not used, but broadcasting turbo stream messages provides it as a local so it needs to be accepted %>
 
 <% conversation = message.conversation %>
 <% conversation_version = streamed ? message.version : @version %>
+<% if message.versions.present?
+  previous_version  = message.versions.push(nil)[ message.versions.index(message.version)-1 ]
+  next_version      = message.versions.push(nil)[ message.versions.index(message.version)+1 ]
+end %>
 <% max_index = conversation.messages.for_conversation_version(conversation_version).count - 1 %>
 <% last_message = message_counter == max_index %>
-<% initial_page_load = last_message && !scroll_down %>
+<% initial_page_load = last_message && !scroll_down # this value is only accurate when rendering last_message %>
 <% scroll_down = scroll_down || initial_page_load %>
 <% thinking ||= last_message && message.content_text == "" && message.not_cancelled? %>
-<% last_message_done = last_message && !thinking && message.content_text.present? %>
-<% debug_tool = false %>
-<%
-if message.versions.present?
-  previous_version  = message.versions.push(nil)[ message.versions.index(message.version)-1 ]
-  next_version      = message.versions.push(nil)[ message.versions.index(message.version)+1 ]
-end
-%>
+<% done_thinking = !thinking && message.content_text.present? %>
+<% debug_tool_messages = false %>
 
 <div
   id="<%= dom_id message %>"
-  class="mb-5 ml-9 sm:ml-3 md:ml-8 group <%= message.only_tool_response? && !debug_tool && "hidden" %>"
   data-role="message"
-  data-timestamp="<%= timestamp %>"
-  data-controller="<%= (scroll_down || only_scroll_down_if_was_bottom) && "message-scroller" %>"
```

Comment:
These stimulus controllers don't need to be on this outer div so I move them down to the `right column` div to clear up this space for some upcoming voice work.